package org.exceptions;

import java.io.EOFException;
import java.io.IOException;

//public class Test {
	//public static void main(String[] args){
		//try{
		 //System.out.printf("1");
		 //int sum = 9 / 0;
		 //System.out.println("2");
		//}
		//catch(ArithmeticException e){
		//System.out.println("3");
		//}
		//catch(Exception e){
		//System.out.println("4");
		//}
		//finally{
		//System.out.println("5");
		 //}
		 //}
		//}
		
//Challenge 2		
	//private void m1(){
		//m2();
		//System.out.println("1");
		//}
		//private void m2(){
		//m3();
		//System.out.println("2");
		//}
		//private void m3(){
		//System.out.println("3");
		//try{
		//int sum = 4/0;
		//System.out.println("4");
		//}
		//catch(ArithmeticException e){
		//System.out.println("5");
		//}
		//System.out.println("7");
		//}
		//public static void main(String[] args){
		//Test obj = new Test();
		//obj.m1();
		//}
		//}
	
//Challenge 3	
	//public static void main(String[] args){
		//try{
		//System.out.println("1");
		//int data = 5 / 0;
		//}
		//catch(ArithmeticException e){
		//System.out.println("2");
		//System.exit(0);
		//}
		//finally{
		//System.out.println("3");
		//}
		//System.out.println("4");
		//}
		//}
	
//Challenge 4
	//public static void main(String[] args){
		//try{
		//System.out.println("1");
		//int data = 5 / 0;
		//}
		//catch(ArithmeticException e){
		//Throwable obj = new Throwable("Sample");
		//try{
		//throw obj;
		//}
		//catch (Throwable e1) {
		//System.out.println("8");
		//}
		//}
		//finally{
		//System.out.println("3");
		//}
		//System.out.println("4");
		//}
		//}
	
//Challenge 5	
	//public static void main(String[] args{
		//try{
		//System.out.println("1");
		//int value = 10 / 0;
		//throw new IOException();
		//}
		//catch(EOFException e){
		//System.out.println("2");
		//}
		//catch(ArithmeticException e){
		//System.out.println("3");
		//}
		//catch(NullPointerException e){
		//System.out.println("4");
		//}
		//catch(IOException e){
		//System.out.println("5");
		//}
		//catch(Exception e){
		//System.out.println("6");
		//}
		//}
		//}
	

	
	
	
	
	


